cpg-library/src/main/java/de/fraunhofer/aisec/cpg/frontends/typescript/ExpressionHandler.kt

```diff
@@ -0,0 +1,243 @@
+    private fun handleStringLiteral(node: TypeScriptNode): Literal<String> {
+        // for now, we also simply parse template expressions as string literals. we
+        // might need a special literal type for that in the future. See
+        // https://github.com/Fraunhofer-AISEC/cpg/issues/463
+        val value =
+            this.lang
+                .getCodeFromRawNode(node)
+                ?.trim()
+                ?.replace("\"", "")
+                ?.replace("`", "")
+                ?.replace("'", "")
+
+        return NodeBuilder.newLiteral(
+            value,
+            TypeParser.createFrom("String", false),
+            lang.getCodeFromRawNode(node)
+        )
+    }
+
+    private fun handleIdentifier(node: TypeScriptNode): Expression {
+        val name = this.lang.getCodeFromRawNode(node)?.trim()
+
+        val ref =
+            NodeBuilder.newDeclaredReferenceExpression(
+                name,
+                UnknownType.getUnknownType(),
+                this.lang.getCodeFromRawNode(node)
+            )
+
+        return ref
+    }
+
+    private fun handlePropertyAccessExpression(node: TypeScriptNode): Expression {
+        val base = this.handle(node.children?.first())
+
+        val name = this.lang.getCodeFromRawNode(node.children?.last())
+
+        val memberExpression =
+            NodeBuilder.newMemberExpression(
+                base,
+                UnknownType.getUnknownType(),
+                name,
+                ".",
+                this.lang.getCodeFromRawNode(node)
+            )
+
+        return memberExpression
+    }
+
+    private fun handleCallExpression(node: TypeScriptNode): Expression {
+        val call: CallExpression
+
+        // peek at the children, to check whether it is a call expression or member call expression
+        val propertyAccess = node.firstChild("PropertyAccessExpression")
+
+        if (propertyAccess != null) {
+            val memberExpression = this.handle(propertyAccess) as MemberExpression
+
+            // we need to build a declared reference expression out of the MemberExpression, because
+            // member calls are not really handled well in the CPG. See
+            // https://github.com/Fraunhofer-AISEC/cpg/issues/298
+            val member =
+                NodeBuilder.newDeclaredReferenceExpression(
+                    memberExpression.name,
+                    memberExpression.type,
+                    memberExpression.name
+                )
+
+            // TODO: fqn - how?
+            val fqn = memberExpression.name
+            call =
+                NodeBuilder.newMemberCallExpression(
+                    memberExpression.name,
+                    fqn,
+                    memberExpression.base,
+                    member,
+                    ".",
+                    this.lang.getCodeFromRawNode(node)
+                )
+        } else {
+            val name = this.lang.getIdentifierName(node)
+
+            // TODO: fqn - how?
```

Comment:
Is this solved? Is the issue that we don't know how to get the fqn or that in the typescript names are hard to map to the fqn concept?

Reply:
Not really solved, for now, I assume `fqn = name`.